code/models/DocumentationPage.php

```diff
@@ -388,6 +390,11 @@ public function setCanonicalUrl($canonicalUrl)
      */
     public function getCanonicalUrl()
     {
+        user_error(
```

Comment:
Perhaps we should use PHPDoc and/or Deprecation instead of user errors

Reply:
If you end up calling these methods, you will be bypassing DocumentationViewer::getCanoncialUrl(), which returns a different results. There's a reasonable chance that your code will be giving unexpected results. My reasoning was that a warning was an appropriate flag to wave to help developers figure out what's happened without halting execution.

An alternative approach would be to make this a major version bump. That also means it won't be immediately foisted on CWP. I wouldn't suggest producing 2.x releases in parallel, though – we have enough to maintain!

If we made a major version bump then I would be inclined to delete this method.

Reply:
Alright, how about adding the PHPDocs for deprecation too on top of the user error? That way IDEs will warn you at least

Reply:
Sounds good. So you'd prefer that approach over a new major releaes?

Reply:
Hah, yeah. We can still remove it in a new major release but at least if it's marked as deprecated we will know to do it when the time comes. I think a new major of this module would be for SS4 compat (#133) if we decide to do it

Reply:
Hey @sminnee if you get a sec would be cool to get this across the line 👍

Reply:
I've updated this - also added a notice to populateCanonicalUrl(), which uses setCanonicalUrl()